zingolib/src/wallet/tx_map/spending_data.rs

```diff
@@ -15,13 +15,16 @@ pub(crate) struct SpendingData {
     witness_trees: WitnessTrees,
     #[getset(get = "pub(crate)", get_mut = "pub(crate)")]
     cached_raw_transactions: HashMap<TxId, Vec<u8>>,
+    #[getset(get = "pub(crate)", get_mut = "pub(crate)")]
+    transparent_ephemeral_ivk: EphemeralIvk,
```

Comment:
can you give some context to what problem are we trying to solve by having keys held anywhere other than UnifiedKeyStore?

Reply:
We don't want to our WalletRead/WalletWrite implementers to need spending keys. One reason is because we want a hardware wallet to be able to call the transaction creation step, without zingo holding the keys, and then passing the transaction to the hardware wallet to sign.

There are a couple places where we need viewing keys for walletread/walletwrite, so we copy those keys into the TxMap.